### ISSUE TITLE
asadmin set command can result in concurrent modification exception

### ISSUE THREAD

**Issue by realityforge** [payara/Payara#277]
Hi,

After upgrading to Payara 4.1.152, the following exception started to appear semi-regularly when configuring the domain with a command not unlike `asadmin set configs.config.server-config.thread-pools.thread-pool.http-thread-pool.max-thread-pool-size=200`

```
[#|2015-05-06T08:56:38.061+1000|SEVERE|Payara 4.1|javax.enterprise.system.core|_ThreadID=43;_ThreadName=admin-listener(3);_TimeMillis=1430866598061;_LevelValue=1000;_MessageID=NCLS-CORE-00003;|Exception while running a command
java.util.ConcurrentModificationException
        at java.util.ArrayList$Itr.checkForComodification(ArrayList.java:859)
        at java.util.ArrayList$Itr.next(ArrayList.java:831)
        at org.jvnet.hk2.config.Transactions.addTransaction(Transactions.java:510)
        at org.jvnet.hk2.config.Transactions.addTransaction(Transactions.java:495)
        at org.jvnet.hk2.config.Transaction.commit(Transaction.java:124)
        at org.jvnet.hk2.config.ConfigSupport.apply(ConfigSupport.java:394)
        at com.sun.enterprise.v3.admin.SetCommand.set(SetCommand.java:481)
        at com.sun.enterprise.v3.admin.SetCommand.execute(SetCommand.java:178)
        at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:539)
        at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:535)
        at java.security.AccessController.doPrivileged(Native Method)
        at javax.security.auth.Subject.doAs(Subject.java:356)
        at com.sun.enterprise.v3.admin.CommandRunnerImpl$2.execute(CommandRunnerImpl.java:534)
        at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:565)
        at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:557)
        at java.security.AccessController.doPrivileged(Native Method)
        at javax.security.auth.Subject.doAs(Subject.java:356)
        at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:556)
        at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:1464)
        at com.sun.enterprise.v3.admin.CommandRunnerImpl.access$1300(CommandRunnerImpl.java:109)
        at com.sun.enterprise.v3.admin.CommandRunnerImpl$ExecutionContext.execute(CommandRunnerImpl.java:1846)
        at com.sun.enterprise.v3.admin.CommandRunnerImpl$ExecutionContext.execute(CommandRunnerImpl.java:1722)
        at org.glassfish.admin.rest.resources.admin.CommandResource.executeCommand(CommandResource.java:404)
        at org.glassfish.admin.rest.resources.admin.CommandResource.execCommandSimpInMultOut(CommandResource.java:234)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606)
```

**Comment by smillidge** [comment]
We are currently running automated tests on asadmin commands. The test builds a domain and configures a number of resources and we don't see the problem.

Are you running multiple asadmin commands concurrently?

**Comment by realityforge** [comment]
Yes we were running concurrent commands when we saw this problem, moving back to single threaded approach has meant that this has not appeared again.

**Comment by OndrejM** [comment]
This should be already fixed in Payara Server 4.1.153, it just slipped from the release notes.
Internally, we track this issue as PAYARA-217.
